auth/libraries/pip-auth/authlib/hashing_handler.py

```diff
@@ -1,25 +1,16 @@
-import base64
-import hashlib
-import secrets
 from typing import Optional
 
+import argon2
+
 
 HASH_WITH_SALT_LENGTH = 152
 
 
 class HashingHandler:
     def hash(self, to_hash: str, hash_reference: Optional[str] = None) -> str:
-        salt = self.__getSalt(hash_reference)
-        hash = hashlib.new("sha512", to_hash.encode())
-        hash.update(salt.encode())
-        hashValue = base64.b64encode(hash.digest()).decode("utf-8")
-        return salt + hashValue
+        ph = argon2.PasswordHasher()
+        return ph.hash(to_hash)
 
-    def __getSalt(self, hash_reference: Optional[str] = None) -> str:
-        if (
-            isinstance(hash_reference, str)
-            and len(hash_reference) == HASH_WITH_SALT_LENGTH
-        ):
-            return hash_reference[0:64]
-        else:
-            return secrets.token_hex(32)
+    def verify(self, hash: str, password: str) -> bool:
+        ph = argon2.PasswordHasher()
+        return ph.verify(hash, password)
```

Comment:
This will not work for legacy passwords, right? In the typescript implementation, I implemented two different cases, one for new argon2 passwords and one for the old password: https://github.com/OpenSlides/openslides-auth-service/blob/19e4f20853a1dde79f89e8045e5fb4bf5a158146/auth/src/api/services/hashing-service.ts#L17
I'm assuming we will have to do that here as well or otherwise it will always return `false` for old passwords.